File: Tech Prototype/sjtu_stranding/src/services/itemService.js
import {postRequest} from "../utils/ajax";
export const getItems = (callback) => {
    const url = "http://localhost:8081/getItems"
    postRequest(url, null, callback);
};
export const getUserItem = (userId, callback) => {
    const data = {userId: userId};
    const url = "http://localhost:8081/getUserItem"
    postRequest(url, data, callback);
};
export const getItem = (itemId, callback) => {
    const data = {itemId: itemId};
    const url = "http://localhost:8081/getItem"
    postRequest(url, data, callback);
};
export const saveItem = (itemId, userId, name, image, number, description, tag1, tag2, tag3, callback) => {
    const data = {itemId: itemId, userId: userId, name: name, image: image, number: number, description: description, tag1: tag1, tag2: tag2, tag3: tag3};
    const url = "http://localhost:8081/saveItem"
    postRequest(url, data, callback);
};

File: Tech Prototype/SJTU_Stranding_backend/src/main/java/com/reins/sjtu_stranding/service/ItemService.java
package com.reins.sjtu_stranding.service;

import com.reins.sjtu_stranding.entity.Item;

import java.util.List;


public interface ItemService {
    List<Item> getItems();
    Item getItem(int itemId);
    List<Item> getUserItem(int userId);

    Item saveItem(int itemId, int userId, String name, String image, int number, String description, String tag1, String tag2, String tag3);

    List<Item> getItemsByX(String x);
}